pkg/azuredisk/azuredisk_v2.go

```diff
@@ -209,26 +203,8 @@ func (d *DriverV2) Run(endpoint, kubeConfigPath string, testBool bool) {
 	cancel()
 }
 
-func (d *DriverV2) checkDiskExists(ctx context.Context, diskURI string) error {
-	diskName, err := GetDiskName(diskURI)
-	if err != nil {
-		return err
-	}
-
-	resourceGroup, err := GetResourceGroupFromURI(diskURI)
-	if err != nil {
-		return err
-	}
-
-	if _, rerr := d.cloud.DisksClient.Get(ctx, resourceGroup, diskName); rerr != nil {
-		return rerr.Error()
-	}
-
-	return nil
-}
-
 func (d *DriverV2) checkDiskCapacity(ctx context.Context, resourceGroup, diskName string, requestGiB int) (bool, error) {
-	disk, err := d.cloud.DisksClient.Get(ctx, resourceGroup, diskName)
+	disk, err := d.controllerProvisioner.Cloud.DisksClient.Get(ctx, resourceGroup, diskName)
```

Comment:
Instead of exposing the Azure client and disk representation, the provider should implement a method that returns an abstract representation of the disk.

Reply:
Moved the functionality to cloud provisioner.